vpr/src/base/region.h

```diff
@@ -0,0 +1,61 @@
+#ifndef REGION_H
+#define REGION_H
+
+#include <vtr_geometry.h>
+
+/**
+ * @file
+ * @brief This file defines the Region class. The Region class stores the data for each constraint region.
+ *
+ * This includes the x and y bounds of the region rectangle and its sub_tile. To lock a block down to a specific location,
+ * when defining the region specify xmin = xmax, ymin = ymax, and specify a subtile value.
+ *
+ */
+
+class Region {
+  public:
+    Region();
```

Comment:
Add @brief on these methods too so they show up in Doxygen (then build Doxygen to the vtr docs and make sure it shows up).